{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b53fb8c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>...</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs3087742</td>\n",
       "      <td>MT:73</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>ENST00000361390</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>34002094,35453788,37998725,37881602</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs3087742</td>\n",
       "      <td>MT:73</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>ENST00000361453</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>34002094,35453788,37998725,37881602</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs3087742</td>\n",
       "      <td>MT:73</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000209082</td>\n",
       "      <td>ENST00000386347</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>34002094,35453788,37998725,37881602</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs3087742</td>\n",
       "      <td>MT:73</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>ENST00000387314</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>34002094,35453788,37998725,37881602</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs3087742</td>\n",
       "      <td>MT:73</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000210077</td>\n",
       "      <td>ENST00000387342</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>34002094,35453788,37998725,37881602</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 82 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Uploaded_variation Location Allele             Gene          Feature  \\\n",
       "0          rs3087742    MT:73      G  ENSG00000198888  ENST00000361390   \n",
       "1          rs3087742    MT:73      G  ENSG00000198763  ENST00000361453   \n",
       "2          rs3087742    MT:73      G  ENSG00000209082  ENST00000386347   \n",
       "3          rs3087742    MT:73      G  ENSG00000210049  ENST00000387314   \n",
       "4          rs3087742    MT:73      G  ENSG00000210077  ENST00000387342   \n",
       "\n",
       "  Feature_type            Consequence cDNA_position CDS_position  \\\n",
       "0   Transcript  upstream_gene_variant             -            -   \n",
       "1   Transcript  upstream_gene_variant             -            -   \n",
       "2   Transcript  upstream_gene_variant             -            -   \n",
       "3   Transcript  upstream_gene_variant             -            -   \n",
       "4   Transcript  upstream_gene_variant             -            -   \n",
       "\n",
       "  Protein_position  ... MAX_AF_POPS CLIN_SIG SOMATIC PHENO  \\\n",
       "0                -  ...           -        -       -     -   \n",
       "1                -  ...           -        -       -     -   \n",
       "2                -  ...           -        -       -     -   \n",
       "3                -  ...           -        -       -     -   \n",
       "4                -  ...           -        -       -     -   \n",
       "\n",
       "                                PUBMED MOTIF_NAME MOTIF_POS HIGH_INF_POS  \\\n",
       "0  34002094,35453788,37998725,37881602          -         -            -   \n",
       "1  34002094,35453788,37998725,37881602          -         -            -   \n",
       "2  34002094,35453788,37998725,37881602          -         -            -   \n",
       "3  34002094,35453788,37998725,37881602          -         -            -   \n",
       "4  34002094,35453788,37998725,37881602          -         -            -   \n",
       "\n",
       "  MOTIF_SCORE_CHANGE TRANSCRIPTION_FACTORS  \n",
       "0                  -                     -  \n",
       "1                  -                     -  \n",
       "2                  -                     -  \n",
       "3                  -                     -  \n",
       "4                  -                     -  \n",
       "\n",
       "[5 rows x 82 columns]"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# File path\n",
    "vep_file = \"/home/subhashree/Project/CTX-Bioinformatics-Intern-Assignment/test2_data_1_vep.tsv\"\n",
    "\n",
    "# Read VEP TSV file and skip '##' metadata lines\n",
    "df_raw = pd.read_csv(\n",
    "    vep_file,\n",
    "    sep=\"\\t\",\n",
    "    comment='##',  # skip metadata lines\n",
    "    engine='python'\n",
    ")\n",
    "\n",
    "# Remove leading '#' from header if present\n",
    "df_raw.rename(columns=lambda x: x.lstrip('#'), inplace=True)\n",
    "\n",
    "# Save cleaned dataframe starting from '#Uploaded_variation'\n",
    "df = df_raw.copy()\n",
    "df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d41abe81",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total variants: 4447820\n",
      "Unique genes: 67066\n"
     ]
    }
   ],
   "source": [
    "# Total variants\n",
    "total_variants = len(df)\n",
    "print(\"Total variants:\", total_variants)\n",
    "\n",
    "# Unique genes\n",
    "unique_genes = df['Gene'].nunique()\n",
    "print(\"Unique genes:\", unique_genes)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b62717b0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique traits/disease conditions: 3\n"
     ]
    }
   ],
   "source": [
    "# Extract traits from PHENO column\n",
    "phenos = df['PHENO'].dropna().astype(str)\n",
    "unique_traits = set()\n",
    "for p in phenos:\n",
    "    for trait in p.replace(';', ',').split(','):\n",
    "        trait = trait.strip()\n",
    "        if trait:\n",
    "            unique_traits.add(trait)\n",
    "\n",
    "num_unique_traits = len(unique_traits)\n",
    "print(\"Unique traits/disease conditions:\", num_unique_traits)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a541d2ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-                                                                                             4433710\n",
       "benign                                                                                           5510\n",
       "likely_benign                                                                                    1601\n",
       "uncertain_significance                                                                           1534\n",
       "pathogenic                                                                                        915\n",
       "                                                                                               ...   \n",
       "pathogenic,pathogenic/likely_pathogenic,not_provided                                                8\n",
       "likely_benign,uncertain_significance,conflicting_interpretations_of_pathogenicity                   8\n",
       "risk_factor,affects,pathogenic,pathogenic/likely_pathogenic,likely_pathogenic,not_provided          2\n",
       "not_provided,likely_benign                                                                          2\n",
       "risk_factor                                                                                         2\n",
       "Name: CLIN_SIG, Length: 97, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"CLIN_SIG\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "3a5c64b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>...</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>rs387906735</td>\n",
       "      <td>MT:608</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>ENST00000361390</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>uncertain_significance,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>11231339</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>rs387906735</td>\n",
       "      <td>MT:608</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>ENST00000361453</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>uncertain_significance,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>11231339</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>rs387906735</td>\n",
       "      <td>MT:608</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000209082</td>\n",
       "      <td>ENST00000386347</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>uncertain_significance,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>11231339</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>rs387906735</td>\n",
       "      <td>MT:608</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>ENST00000387314</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>32</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>uncertain_significance,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>11231339</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287</th>\n",
       "      <td>rs387906735</td>\n",
       "      <td>MT:608</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000210077</td>\n",
       "      <td>ENST00000387342</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>uncertain_significance,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>11231339</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4301254</th>\n",
       "      <td>rs1555050</td>\n",
       "      <td>22:50627253</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000289244</td>\n",
       "      <td>ENST00000806467</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADe_AFR,gnomADe_AMR,gnomADe_ASJ,gnomADe_EA...</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4301255</th>\n",
       "      <td>rs1555050</td>\n",
       "      <td>22:50627253</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000289244</td>\n",
       "      <td>ENST00000806468</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADe_AFR,gnomADe_AMR,gnomADe_ASJ,gnomADe_EA...</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4301256</th>\n",
       "      <td>rs1555050</td>\n",
       "      <td>22:50627253</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000289244</td>\n",
       "      <td>ENST00000806469</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADe_AFR,gnomADe_AMR,gnomADe_ASJ,gnomADe_EA...</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4301257</th>\n",
       "      <td>rs1555050</td>\n",
       "      <td>22:50627253</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000289244</td>\n",
       "      <td>ENST00000806470</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADe_AFR,gnomADe_AMR,gnomADe_ASJ,gnomADe_EA...</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4301258</th>\n",
       "      <td>rs1555050</td>\n",
       "      <td>22:50627253</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000289244</td>\n",
       "      <td>ENST00000806471</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADe_AFR,gnomADe_AMR,gnomADe_ASJ,gnomADe_EA...</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3525 rows Ã— 82 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Uploaded_variation     Location Allele             Gene  \\\n",
       "283            rs387906735       MT:608      G  ENSG00000198888   \n",
       "284            rs387906735       MT:608      G  ENSG00000198763   \n",
       "285            rs387906735       MT:608      G  ENSG00000209082   \n",
       "286            rs387906735       MT:608      G  ENSG00000210049   \n",
       "287            rs387906735       MT:608      G  ENSG00000210077   \n",
       "...                    ...          ...    ...              ...   \n",
       "4301254          rs1555050  22:50627253      T  ENSG00000289244   \n",
       "4301255          rs1555050  22:50627253      T  ENSG00000289244   \n",
       "4301256          rs1555050  22:50627253      T  ENSG00000289244   \n",
       "4301257          rs1555050  22:50627253      T  ENSG00000289244   \n",
       "4301258          rs1555050  22:50627253      T  ENSG00000289244   \n",
       "\n",
       "                 Feature Feature_type                         Consequence  \\\n",
       "283      ENST00000361390   Transcript               upstream_gene_variant   \n",
       "284      ENST00000361453   Transcript               upstream_gene_variant   \n",
       "285      ENST00000386347   Transcript               upstream_gene_variant   \n",
       "286      ENST00000387314   Transcript  non_coding_transcript_exon_variant   \n",
       "287      ENST00000387342   Transcript               upstream_gene_variant   \n",
       "...                  ...          ...                                 ...   \n",
       "4301254  ENST00000806467   Transcript               upstream_gene_variant   \n",
       "4301255  ENST00000806468   Transcript               upstream_gene_variant   \n",
       "4301256  ENST00000806469   Transcript               upstream_gene_variant   \n",
       "4301257  ENST00000806470   Transcript               upstream_gene_variant   \n",
       "4301258  ENST00000806471   Transcript               upstream_gene_variant   \n",
       "\n",
       "        cDNA_position CDS_position Protein_position  ...  \\\n",
       "283                 -            -                -  ...   \n",
       "284                 -            -                -  ...   \n",
       "285                 -            -                -  ...   \n",
       "286                32            -                -  ...   \n",
       "287                 -            -                -  ...   \n",
       "...               ...          ...              ...  ...   \n",
       "4301254             -            -                -  ...   \n",
       "4301255             -            -                -  ...   \n",
       "4301256             -            -                -  ...   \n",
       "4301257             -            -                -  ...   \n",
       "4301258             -            -                -  ...   \n",
       "\n",
       "                                               MAX_AF_POPS  \\\n",
       "283                                                      -   \n",
       "284                                                      -   \n",
       "285                                                      -   \n",
       "286                                                      -   \n",
       "287                                                      -   \n",
       "...                                                    ...   \n",
       "4301254  gnomADe_AFR,gnomADe_AMR,gnomADe_ASJ,gnomADe_EA...   \n",
       "4301255  gnomADe_AFR,gnomADe_AMR,gnomADe_ASJ,gnomADe_EA...   \n",
       "4301256  gnomADe_AFR,gnomADe_AMR,gnomADe_ASJ,gnomADe_EA...   \n",
       "4301257  gnomADe_AFR,gnomADe_AMR,gnomADe_ASJ,gnomADe_EA...   \n",
       "4301258  gnomADe_AFR,gnomADe_AMR,gnomADe_ASJ,gnomADe_EA...   \n",
       "\n",
       "                                  CLIN_SIG SOMATIC PHENO    PUBMED MOTIF_NAME  \\\n",
       "283      uncertain_significance,pathogenic       -     1  11231339          -   \n",
       "284      uncertain_significance,pathogenic       -     1  11231339          -   \n",
       "285      uncertain_significance,pathogenic       -     1  11231339          -   \n",
       "286      uncertain_significance,pathogenic       -     1  11231339          -   \n",
       "287      uncertain_significance,pathogenic       -     1  11231339          -   \n",
       "...                                    ...     ...   ...       ...        ...   \n",
       "4301254                         pathogenic     0,1   1,1         -          -   \n",
       "4301255                         pathogenic     0,1   1,1         -          -   \n",
       "4301256                         pathogenic     0,1   1,1         -          -   \n",
       "4301257                         pathogenic     0,1   1,1         -          -   \n",
       "4301258                         pathogenic     0,1   1,1         -          -   \n",
       "\n",
       "        MOTIF_POS HIGH_INF_POS MOTIF_SCORE_CHANGE TRANSCRIPTION_FACTORS  \n",
       "283             -            -                  -                     -  \n",
       "284             -            -                  -                     -  \n",
       "285             -            -                  -                     -  \n",
       "286             -            -                  -                     -  \n",
       "287             -            -                  -                     -  \n",
       "...           ...          ...                ...                   ...  \n",
       "4301254         -            -                  -                     -  \n",
       "4301255         -            -                  -                     -  \n",
       "4301256         -            -                  -                     -  \n",
       "4301257         -            -                  -                     -  \n",
       "4301258         -            -                  -                     -  \n",
       "\n",
       "[3525 rows x 82 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_patho = df[df[\"CLIN_SIG\"].str.contains(\"pathogenic\", case=False, na=False)]\n",
    "df_patho"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4a108ba8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>...</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2501</th>\n",
       "      <td>rs199474657</td>\n",
       "      <td>MT:3243</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>ENST00000361390</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>likely_pathogenic,pathogenic,not_provided,path...</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,33785019,16326995,26076356,20610441,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2502</th>\n",
       "      <td>rs199474657</td>\n",
       "      <td>MT:3243</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>ENST00000361453</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>likely_pathogenic,pathogenic,not_provided,path...</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,33785019,16326995,26076356,20610441,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2503</th>\n",
       "      <td>rs199474657</td>\n",
       "      <td>MT:3243</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000198804</td>\n",
       "      <td>ENST00000361624</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>likely_pathogenic,pathogenic,not_provided,path...</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,33785019,16326995,26076356,20610441,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2504</th>\n",
       "      <td>rs199474657</td>\n",
       "      <td>MT:3243</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000198712</td>\n",
       "      <td>ENST00000361739</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>likely_pathogenic,pathogenic,not_provided,path...</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,33785019,16326995,26076356,20610441,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2505</th>\n",
       "      <td>rs199474657</td>\n",
       "      <td>MT:3243</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000209082</td>\n",
       "      <td>ENST00000386347</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>14</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>likely_pathogenic,pathogenic,not_provided,path...</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>25741868,33785019,16326995,26076356,20610441,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3941093</th>\n",
       "      <td>rs886037859</td>\n",
       "      <td>19:1220449</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000118046</td>\n",
       "      <td>ENST00000714323</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1669</td>\n",
       "      <td>541</td>\n",
       "      <td>181</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0,1,1,1</td>\n",
       "      <td>1,1,1,1</td>\n",
       "      <td>25741868</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3948139</th>\n",
       "      <td>rs121434498</td>\n",
       "      <td>19:4117551</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000126934</td>\n",
       "      <td>ENST00000262948</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>418</td>\n",
       "      <td>171</td>\n",
       "      <td>57</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADe_ASJ</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0,1,1,1,1</td>\n",
       "      <td>1,1,1,1,1</td>\n",
       "      <td>26619011,18456719,16439621,18042262,19156172</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3948140</th>\n",
       "      <td>rs121434498</td>\n",
       "      <td>19:4117551</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000126934</td>\n",
       "      <td>ENST00000394867</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>610</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADe_ASJ</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0,1,1,1,1</td>\n",
       "      <td>1,1,1,1,1</td>\n",
       "      <td>26619011,18456719,16439621,18042262,19156172</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3948141</th>\n",
       "      <td>rs121434498</td>\n",
       "      <td>19:4117551</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000126934</td>\n",
       "      <td>ENST00000599345</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>368</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADe_ASJ</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0,1,1,1,1</td>\n",
       "      <td>1,1,1,1,1</td>\n",
       "      <td>26619011,18456719,16439621,18042262,19156172</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3948142</th>\n",
       "      <td>rs121434498</td>\n",
       "      <td>19:4117551</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000126934</td>\n",
       "      <td>ENST00000687128</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>610</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADe_ASJ</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0,1,1,1,1</td>\n",
       "      <td>1,1,1,1,1</td>\n",
       "      <td>26619011,18456719,16439621,18042262,19156172</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1251 rows Ã— 82 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Uploaded_variation    Location Allele             Gene  \\\n",
       "2501           rs199474657     MT:3243      G  ENSG00000198888   \n",
       "2502           rs199474657     MT:3243      G  ENSG00000198763   \n",
       "2503           rs199474657     MT:3243      G  ENSG00000198804   \n",
       "2504           rs199474657     MT:3243      G  ENSG00000198712   \n",
       "2505           rs199474657     MT:3243      G  ENSG00000209082   \n",
       "...                    ...         ...    ...              ...   \n",
       "3941093        rs886037859  19:1220449      G  ENSG00000118046   \n",
       "3948139        rs121434498  19:4117551      C  ENSG00000126934   \n",
       "3948140        rs121434498  19:4117551      C  ENSG00000126934   \n",
       "3948141        rs121434498  19:4117551      C  ENSG00000126934   \n",
       "3948142        rs121434498  19:4117551      C  ENSG00000126934   \n",
       "\n",
       "                 Feature Feature_type                         Consequence  \\\n",
       "2501     ENST00000361390   Transcript               upstream_gene_variant   \n",
       "2502     ENST00000361453   Transcript               upstream_gene_variant   \n",
       "2503     ENST00000361624   Transcript               upstream_gene_variant   \n",
       "2504     ENST00000361739   Transcript               upstream_gene_variant   \n",
       "2505     ENST00000386347   Transcript  non_coding_transcript_exon_variant   \n",
       "...                  ...          ...                                 ...   \n",
       "3941093  ENST00000714323   Transcript                    missense_variant   \n",
       "3948139  ENST00000262948   Transcript                    missense_variant   \n",
       "3948140  ENST00000394867   Transcript  non_coding_transcript_exon_variant   \n",
       "3948141  ENST00000599345   Transcript  non_coding_transcript_exon_variant   \n",
       "3948142  ENST00000687128   Transcript  non_coding_transcript_exon_variant   \n",
       "\n",
       "        cDNA_position CDS_position Protein_position  ...  MAX_AF_POPS  \\\n",
       "2501                -            -                -  ...            -   \n",
       "2502                -            -                -  ...            -   \n",
       "2503                -            -                -  ...            -   \n",
       "2504                -            -                -  ...            -   \n",
       "2505               14            -                -  ...            -   \n",
       "...               ...          ...              ...  ...          ...   \n",
       "3941093          1669          541              181  ...            -   \n",
       "3948139           418          171               57  ...  gnomADe_ASJ   \n",
       "3948140           610            -                -  ...  gnomADe_ASJ   \n",
       "3948141           368            -                -  ...  gnomADe_ASJ   \n",
       "3948142           610            -                -  ...  gnomADe_ASJ   \n",
       "\n",
       "                                                  CLIN_SIG    SOMATIC  \\\n",
       "2501     likely_pathogenic,pathogenic,not_provided,path...          -   \n",
       "2502     likely_pathogenic,pathogenic,not_provided,path...          -   \n",
       "2503     likely_pathogenic,pathogenic,not_provided,path...          -   \n",
       "2504     likely_pathogenic,pathogenic,not_provided,path...          -   \n",
       "2505     likely_pathogenic,pathogenic,not_provided,path...          -   \n",
       "...                                                    ...        ...   \n",
       "3941093                                  likely_pathogenic    0,1,1,1   \n",
       "3948139                                  likely_pathogenic  0,1,1,1,1   \n",
       "3948140                                  likely_pathogenic  0,1,1,1,1   \n",
       "3948141                                  likely_pathogenic  0,1,1,1,1   \n",
       "3948142                                  likely_pathogenic  0,1,1,1,1   \n",
       "\n",
       "             PHENO                                             PUBMED  \\\n",
       "2501             1  25741868,33785019,16326995,26076356,20610441,2...   \n",
       "2502             1  25741868,33785019,16326995,26076356,20610441,2...   \n",
       "2503             1  25741868,33785019,16326995,26076356,20610441,2...   \n",
       "2504             1  25741868,33785019,16326995,26076356,20610441,2...   \n",
       "2505             1  25741868,33785019,16326995,26076356,20610441,2...   \n",
       "...            ...                                                ...   \n",
       "3941093    1,1,1,1                                           25741868   \n",
       "3948139  1,1,1,1,1       26619011,18456719,16439621,18042262,19156172   \n",
       "3948140  1,1,1,1,1       26619011,18456719,16439621,18042262,19156172   \n",
       "3948141  1,1,1,1,1       26619011,18456719,16439621,18042262,19156172   \n",
       "3948142  1,1,1,1,1       26619011,18456719,16439621,18042262,19156172   \n",
       "\n",
       "        MOTIF_NAME MOTIF_POS HIGH_INF_POS MOTIF_SCORE_CHANGE  \\\n",
       "2501             -         -            -                  -   \n",
       "2502             -         -            -                  -   \n",
       "2503             -         -            -                  -   \n",
       "2504             -         -            -                  -   \n",
       "2505             -         -            -                  -   \n",
       "...            ...       ...          ...                ...   \n",
       "3941093          -         -            -                  -   \n",
       "3948139          -         -            -                  -   \n",
       "3948140          -         -            -                  -   \n",
       "3948141          -         -            -                  -   \n",
       "3948142          -         -            -                  -   \n",
       "\n",
       "        TRANSCRIPTION_FACTORS  \n",
       "2501                        -  \n",
       "2502                        -  \n",
       "2503                        -  \n",
       "2504                        -  \n",
       "2505                        -  \n",
       "...                       ...  \n",
       "3941093                     -  \n",
       "3948139                     -  \n",
       "3948140                     -  \n",
       "3948141                     -  \n",
       "3948142                     -  \n",
       "\n",
       "[1251 rows x 82 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_likely_patho = df[df[\"CLIN_SIG\"].str.contains(\"likely_pathogenic\", case=False, na=False)]\n",
    "df_likely_patho"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "31fce2c1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variants with pathogenicity: 196\n"
     ]
    }
   ],
   "source": [
    "print(\"Variants with pathogenicity:\", df_patho[\"Uploaded_variation\"].nunique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "accec8d0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variants with likely pathogenicity: 73\n"
     ]
    }
   ],
   "source": [
    "print(\"Variants with likely pathogenicity:\", df_likely_patho[\"Uploaded_variation\"].nunique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "843a2e95",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[1], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m#No. of variants per chromosome\u001b[39;00m\n\u001b[0;32m----> 2\u001b[0m df[[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mchr\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mposition\u001b[39m\u001b[38;5;124m'\u001b[39m]] \u001b[38;5;241m=\u001b[39m \u001b[43mdf\u001b[49m[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mLocation\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39mstr\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m:\u001b[39m\u001b[38;5;124m'\u001b[39m, expand\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[1;32m      3\u001b[0m df[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mchr\u001b[39m\u001b[38;5;124m\"\u001b[39m]\u001b[38;5;241m.\u001b[39mvalue_counts()\n",
      "\u001b[0;31mNameError\u001b[0m: name 'df' is not defined"
     ]
    }
   ],
   "source": [
    "#No. of variants per chromosome\n",
    "df[['chr', 'position']] = df['Location'].str.split(':', expand=True)\n",
    "df[\"chr\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5e3b48ec",
   "metadata": {},
   "outputs": [
    {
     "ename": "",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31mThe Kernel crashed while executing code in the current cell or a previous cell. \n",
      "\u001b[1;31mPlease review the code in the cell(s) to identify a possible cause of the failure. \n",
      "\u001b[1;31mClick <a href='https://aka.ms/vscodeJupyterKernelCrash'>here</a> for more info. \n",
      "\u001b[1;31mView Jupyter <a href='command:jupyter.viewOutput'>log</a> for further details."
     ]
    }
   ],
   "source": [
    "# Identify rows that do NOT start with 'rs'\n",
    "no_rs = df[~df['Uploaded_variation'].str.startswith('rs', na=False)]\n",
    "\n",
    "# Optional: rows with rsIDs\n",
    "with_rs = df[df['Uploaded_variation'].str.startswith('rs', na=False)]\n",
    "\n",
    "# Print or save\n",
    "print(\"Variants without rsIDs:\")\n",
    "no_rs[\"Uploaded_variation\"].nunique()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4b5c6c00",
   "metadata": {},
   "outputs": [],
   "source": [
    "print(\"Variants with rsIDs:\")\n",
    "with_rs[\"Uploaded_variation\"].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "480baee6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "qiime2-amplicon-2023.9",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
