{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "b53fb8c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>...</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1_871334_G/T</td>\n",
       "      <td>1:871334</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000234711</td>\n",
       "      <td>ENST00000415481</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1_871334_G/T</td>\n",
       "      <td>1:871334</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000230368</td>\n",
       "      <td>ENST00000427857</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1_871334_G/T</td>\n",
       "      <td>1:871334</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000230368</td>\n",
       "      <td>ENST00000432963</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1_871334_G/T</td>\n",
       "      <td>1:871334</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000230368</td>\n",
       "      <td>ENST00000446136</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1_871334_G/T</td>\n",
       "      <td>1:871334</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000283040</td>\n",
       "      <td>ENST00000635557</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 82 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Uploaded_variation  Location Allele             Gene          Feature  \\\n",
       "0       1_871334_G/T  1:871334      T  ENSG00000234711  ENST00000415481   \n",
       "1       1_871334_G/T  1:871334      T  ENSG00000230368  ENST00000427857   \n",
       "2       1_871334_G/T  1:871334      T  ENSG00000230368  ENST00000432963   \n",
       "3       1_871334_G/T  1:871334      T  ENSG00000230368  ENST00000446136   \n",
       "4       1_871334_G/T  1:871334      T  ENSG00000283040  ENST00000635557   \n",
       "\n",
       "  Feature_type                                   Consequence cDNA_position  \\\n",
       "0   Transcript                         upstream_gene_variant             -   \n",
       "1   Transcript  intron_variant,non_coding_transcript_variant             -   \n",
       "2   Transcript                         upstream_gene_variant             -   \n",
       "3   Transcript  intron_variant,non_coding_transcript_variant             -   \n",
       "4   Transcript                       downstream_gene_variant             -   \n",
       "\n",
       "  CDS_position Protein_position  ... MAX_AF_POPS CLIN_SIG SOMATIC PHENO  \\\n",
       "0            -                -  ...           -        -       -     -   \n",
       "1            -                -  ...           -        -       -     -   \n",
       "2            -                -  ...           -        -       -     -   \n",
       "3            -                -  ...           -        -       -     -   \n",
       "4            -                -  ...           -        -       -     -   \n",
       "\n",
       "  PUBMED MOTIF_NAME MOTIF_POS HIGH_INF_POS MOTIF_SCORE_CHANGE  \\\n",
       "0      -          -         -            -                  -   \n",
       "1      -          -         -            -                  -   \n",
       "2      -          -         -            -                  -   \n",
       "3      -          -         -            -                  -   \n",
       "4      -          -         -            -                  -   \n",
       "\n",
       "  TRANSCRIPTION_FACTORS  \n",
       "0                     -  \n",
       "1                     -  \n",
       "2                     -  \n",
       "3                     -  \n",
       "4                     -  \n",
       "\n",
       "[5 rows x 82 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# File path\n",
    "vep_file = \"/home/subhashree/Project/CTX-Bioinformatics-Intern-Assignment/test1_data_1_vep.tsv\"\n",
    "\n",
    "# Read VEP TSV file and skip '##' metadata lines\n",
    "df_raw = pd.read_csv(\n",
    "    vep_file,\n",
    "    sep=\"\\t\",\n",
    "    comment='##',  # skip metadata lines\n",
    "    engine='python'\n",
    ")\n",
    "\n",
    "# Remove leading '#' from header if present\n",
    "df_raw.rename(columns=lambda x: x.lstrip('#'), inplace=True)\n",
    "\n",
    "# Save cleaned dataframe starting from '#Uploaded_variation'\n",
    "df = df_raw.copy()\n",
    "df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d41abe81",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total variants: 228055\n",
      "Unique genes: 14674\n"
     ]
    }
   ],
   "source": [
    "# Total variants\n",
    "total_variants = len(df)\n",
    "print(\"Total variants:\", total_variants)\n",
    "\n",
    "# Unique genes\n",
    "unique_genes = df['Gene'].nunique()\n",
    "print(\"Unique genes:\", unique_genes)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "b62717b0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique traits/disease conditions: 3\n"
     ]
    }
   ],
   "source": [
    "# Extract traits from PHENO column\n",
    "phenos = df['PHENO'].dropna().astype(str)\n",
    "unique_traits = set()\n",
    "for p in phenos:\n",
    "    for trait in p.replace(';', ',').split(','):\n",
    "        trait = trait.strip()\n",
    "        if trait:\n",
    "            unique_traits.add(trait)\n",
    "\n",
    "num_unique_traits = len(unique_traits)\n",
    "print(\"Unique traits/disease conditions:\", num_unique_traits)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "a541d2ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-                                                                    226521\n",
       "benign                                                                 1078\n",
       "uncertain_significance                                                   98\n",
       "likely_benign,benign                                                     98\n",
       "likely_benign                                                            89\n",
       "benign,likely_benign                                                     61\n",
       "not_provided,benign                                                      54\n",
       "benign,not_provided                                                      27\n",
       "benign/likely_benign,benign                                              16\n",
       "likely_benign,conflicting_interpretations_of_pathogenicity,benign        13\n",
       "Name: CLIN_SIG, dtype: int64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"CLIN_SIG\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "3a5c64b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>...</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>116551</th>\n",
       "      <td>11_534242_A/G</td>\n",
       "      <td>11:534242</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000161328</td>\n",
       "      <td>ENST00000270115</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>0,1,1</td>\n",
       "      <td>1,1,1</td>\n",
       "      <td>25741868,24033266,33390813,23757202,20569235,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116552</th>\n",
       "      <td>11_534242_A/G</td>\n",
       "      <td>11:534242</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000174775</td>\n",
       "      <td>ENST00000311189</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>295</td>\n",
       "      <td>81</td>\n",
       "      <td>27</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>0,1,1</td>\n",
       "      <td>1,1,1</td>\n",
       "      <td>25741868,24033266,33390813,23757202,20569235,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116553</th>\n",
       "      <td>11_534242_A/G</td>\n",
       "      <td>11:534242</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000174775</td>\n",
       "      <td>ENST00000397594</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>134</td>\n",
       "      <td>81</td>\n",
       "      <td>27</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>0,1,1</td>\n",
       "      <td>1,1,1</td>\n",
       "      <td>25741868,24033266,33390813,23757202,20569235,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116554</th>\n",
       "      <td>11_534242_A/G</td>\n",
       "      <td>11:534242</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000174775</td>\n",
       "      <td>ENST00000397596</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>217</td>\n",
       "      <td>81</td>\n",
       "      <td>27</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>0,1,1</td>\n",
       "      <td>1,1,1</td>\n",
       "      <td>25741868,24033266,33390813,23757202,20569235,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116555</th>\n",
       "      <td>11_534242_A/G</td>\n",
       "      <td>11:534242</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000174775</td>\n",
       "      <td>ENST00000417302</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>295</td>\n",
       "      <td>81</td>\n",
       "      <td>27</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>0,1,1</td>\n",
       "      <td>1,1,1</td>\n",
       "      <td>25741868,24033266,33390813,23757202,20569235,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116556</th>\n",
       "      <td>11_534242_A/G</td>\n",
       "      <td>11:534242</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000174775</td>\n",
       "      <td>ENST00000451590</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>269</td>\n",
       "      <td>81</td>\n",
       "      <td>27</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>0,1,1</td>\n",
       "      <td>1,1,1</td>\n",
       "      <td>25741868,24033266,33390813,23757202,20569235,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116557</th>\n",
       "      <td>11_534242_A/G</td>\n",
       "      <td>11:534242</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000174775</td>\n",
       "      <td>ENST00000462734</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant,NMD_transcript_variant</td>\n",
       "      <td>1120</td>\n",
       "      <td>81</td>\n",
       "      <td>27</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>0,1,1</td>\n",
       "      <td>1,1,1</td>\n",
       "      <td>25741868,24033266,33390813,23757202,20569235,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116558</th>\n",
       "      <td>11_534242_A/G</td>\n",
       "      <td>11:534242</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000174775</td>\n",
       "      <td>ENST00000468682</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>569</td>\n",
       "      <td>81</td>\n",
       "      <td>27</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>0,1,1</td>\n",
       "      <td>1,1,1</td>\n",
       "      <td>25741868,24033266,33390813,23757202,20569235,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116559</th>\n",
       "      <td>11_534242_A/G</td>\n",
       "      <td>11:534242</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000174775</td>\n",
       "      <td>ENST00000479482</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>0,1,1</td>\n",
       "      <td>1,1,1</td>\n",
       "      <td>25741868,24033266,33390813,23757202,20569235,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116560</th>\n",
       "      <td>11_534242_A/G</td>\n",
       "      <td>11:534242</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000174775</td>\n",
       "      <td>ENST00000482021</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>0,1,1</td>\n",
       "      <td>1,1,1</td>\n",
       "      <td>25741868,24033266,33390813,23757202,20569235,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116561</th>\n",
       "      <td>11_534242_A/G</td>\n",
       "      <td>11:534242</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000174775</td>\n",
       "      <td>ENST00000493230</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant,NMD_transcript_variant</td>\n",
       "      <td>257</td>\n",
       "      <td>81</td>\n",
       "      <td>27</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>0,1,1</td>\n",
       "      <td>1,1,1</td>\n",
       "      <td>25741868,24033266,33390813,23757202,20569235,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116562</th>\n",
       "      <td>11_534242_A/G</td>\n",
       "      <td>11:534242</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000254739</td>\n",
       "      <td>ENST00000526431</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>0,1,1</td>\n",
       "      <td>1,1,1</td>\n",
       "      <td>25741868,24033266,33390813,23757202,20569235,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116563</th>\n",
       "      <td>11_534242_A/G</td>\n",
       "      <td>11:534242</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000289997</td>\n",
       "      <td>ENST00000702393</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>0,1,1</td>\n",
       "      <td>1,1,1</td>\n",
       "      <td>25741868,24033266,33390813,23757202,20569235,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13 rows Ã— 82 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Uploaded_variation   Location Allele             Gene          Feature  \\\n",
       "116551      11_534242_A/G  11:534242      G  ENSG00000161328  ENST00000270115   \n",
       "116552      11_534242_A/G  11:534242      G  ENSG00000174775  ENST00000311189   \n",
       "116553      11_534242_A/G  11:534242      G  ENSG00000174775  ENST00000397594   \n",
       "116554      11_534242_A/G  11:534242      G  ENSG00000174775  ENST00000397596   \n",
       "116555      11_534242_A/G  11:534242      G  ENSG00000174775  ENST00000417302   \n",
       "116556      11_534242_A/G  11:534242      G  ENSG00000174775  ENST00000451590   \n",
       "116557      11_534242_A/G  11:534242      G  ENSG00000174775  ENST00000462734   \n",
       "116558      11_534242_A/G  11:534242      G  ENSG00000174775  ENST00000468682   \n",
       "116559      11_534242_A/G  11:534242      G  ENSG00000174775  ENST00000479482   \n",
       "116560      11_534242_A/G  11:534242      G  ENSG00000174775  ENST00000482021   \n",
       "116561      11_534242_A/G  11:534242      G  ENSG00000174775  ENST00000493230   \n",
       "116562      11_534242_A/G  11:534242      G  ENSG00000254739  ENST00000526431   \n",
       "116563      11_534242_A/G  11:534242      G  ENSG00000289997  ENST00000702393   \n",
       "\n",
       "       Feature_type                                Consequence cDNA_position  \\\n",
       "116551   Transcript                      upstream_gene_variant             -   \n",
       "116552   Transcript                         synonymous_variant           295   \n",
       "116553   Transcript                         synonymous_variant           134   \n",
       "116554   Transcript                         synonymous_variant           217   \n",
       "116555   Transcript                         synonymous_variant           295   \n",
       "116556   Transcript                         synonymous_variant           269   \n",
       "116557   Transcript  synonymous_variant,NMD_transcript_variant          1120   \n",
       "116558   Transcript                         synonymous_variant           569   \n",
       "116559   Transcript                      upstream_gene_variant             -   \n",
       "116560   Transcript                    downstream_gene_variant             -   \n",
       "116561   Transcript  synonymous_variant,NMD_transcript_variant           257   \n",
       "116562   Transcript                    downstream_gene_variant             -   \n",
       "116563   Transcript                    downstream_gene_variant             -   \n",
       "\n",
       "       CDS_position Protein_position  ...  MAX_AF_POPS  \\\n",
       "116551            -                -  ...  gnomADg_AFR   \n",
       "116552           81               27  ...  gnomADg_AFR   \n",
       "116553           81               27  ...  gnomADg_AFR   \n",
       "116554           81               27  ...  gnomADg_AFR   \n",
       "116555           81               27  ...  gnomADg_AFR   \n",
       "116556           81               27  ...  gnomADg_AFR   \n",
       "116557           81               27  ...  gnomADg_AFR   \n",
       "116558           81               27  ...  gnomADg_AFR   \n",
       "116559            -                -  ...  gnomADg_AFR   \n",
       "116560            -                -  ...  gnomADg_AFR   \n",
       "116561           81               27  ...  gnomADg_AFR   \n",
       "116562            -                -  ...  gnomADg_AFR   \n",
       "116563            -                -  ...  gnomADg_AFR   \n",
       "\n",
       "                                                 CLIN_SIG SOMATIC  PHENO  \\\n",
       "116551  likely_benign,conflicting_interpretations_of_p...   0,1,1  1,1,1   \n",
       "116552  likely_benign,conflicting_interpretations_of_p...   0,1,1  1,1,1   \n",
       "116553  likely_benign,conflicting_interpretations_of_p...   0,1,1  1,1,1   \n",
       "116554  likely_benign,conflicting_interpretations_of_p...   0,1,1  1,1,1   \n",
       "116555  likely_benign,conflicting_interpretations_of_p...   0,1,1  1,1,1   \n",
       "116556  likely_benign,conflicting_interpretations_of_p...   0,1,1  1,1,1   \n",
       "116557  likely_benign,conflicting_interpretations_of_p...   0,1,1  1,1,1   \n",
       "116558  likely_benign,conflicting_interpretations_of_p...   0,1,1  1,1,1   \n",
       "116559  likely_benign,conflicting_interpretations_of_p...   0,1,1  1,1,1   \n",
       "116560  likely_benign,conflicting_interpretations_of_p...   0,1,1  1,1,1   \n",
       "116561  likely_benign,conflicting_interpretations_of_p...   0,1,1  1,1,1   \n",
       "116562  likely_benign,conflicting_interpretations_of_p...   0,1,1  1,1,1   \n",
       "116563  likely_benign,conflicting_interpretations_of_p...   0,1,1  1,1,1   \n",
       "\n",
       "                                                   PUBMED MOTIF_NAME  \\\n",
       "116551  25741868,24033266,33390813,23757202,20569235,2...          -   \n",
       "116552  25741868,24033266,33390813,23757202,20569235,2...          -   \n",
       "116553  25741868,24033266,33390813,23757202,20569235,2...          -   \n",
       "116554  25741868,24033266,33390813,23757202,20569235,2...          -   \n",
       "116555  25741868,24033266,33390813,23757202,20569235,2...          -   \n",
       "116556  25741868,24033266,33390813,23757202,20569235,2...          -   \n",
       "116557  25741868,24033266,33390813,23757202,20569235,2...          -   \n",
       "116558  25741868,24033266,33390813,23757202,20569235,2...          -   \n",
       "116559  25741868,24033266,33390813,23757202,20569235,2...          -   \n",
       "116560  25741868,24033266,33390813,23757202,20569235,2...          -   \n",
       "116561  25741868,24033266,33390813,23757202,20569235,2...          -   \n",
       "116562  25741868,24033266,33390813,23757202,20569235,2...          -   \n",
       "116563  25741868,24033266,33390813,23757202,20569235,2...          -   \n",
       "\n",
       "       MOTIF_POS HIGH_INF_POS MOTIF_SCORE_CHANGE TRANSCRIPTION_FACTORS  \n",
       "116551         -            -                  -                     -  \n",
       "116552         -            -                  -                     -  \n",
       "116553         -            -                  -                     -  \n",
       "116554         -            -                  -                     -  \n",
       "116555         -            -                  -                     -  \n",
       "116556         -            -                  -                     -  \n",
       "116557         -            -                  -                     -  \n",
       "116558         -            -                  -                     -  \n",
       "116559         -            -                  -                     -  \n",
       "116560         -            -                  -                     -  \n",
       "116561         -            -                  -                     -  \n",
       "116562         -            -                  -                     -  \n",
       "116563         -            -                  -                     -  \n",
       "\n",
       "[13 rows x 82 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_patho = df[df[\"CLIN_SIG\"].str.contains(\"pathogenic\", case=False, na=False)]\n",
    "df_patho"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "4a108ba8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>...</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows Ã— 82 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Uploaded_variation, Location, Allele, Gene, Feature, Feature_type, Consequence, cDNA_position, CDS_position, Protein_position, Amino_acids, Codons, Existing_variation, IMPACT, DISTANCE, STRAND, FLAGS, VARIANT_CLASS, SYMBOL, SYMBOL_SOURCE, HGNC_ID, BIOTYPE, CANONICAL, MANE, MANE_SELECT, MANE_PLUS_CLINICAL, TSL, APPRIS, CCDS, ENSP, SWISSPROT, TREMBL, UNIPARC, UNIPROT_ISOFORM, GENE_PHENO, SIFT, PolyPhen, EXON, INTRON, DOMAINS, miRNA, HGVSc, HGVSp, HGVS_OFFSET, AF, AFR_AF, AMR_AF, EAS_AF, EUR_AF, SAS_AF, gnomADe_AF, gnomADe_AFR_AF, gnomADe_AMR_AF, gnomADe_ASJ_AF, gnomADe_EAS_AF, gnomADe_FIN_AF, gnomADe_MID_AF, gnomADe_NFE_AF, gnomADe_REMAINING_AF, gnomADe_SAS_AF, gnomADg_AF, gnomADg_AFR_AF, gnomADg_AMI_AF, gnomADg_AMR_AF, gnomADg_ASJ_AF, gnomADg_EAS_AF, gnomADg_FIN_AF, gnomADg_MID_AF, gnomADg_NFE_AF, gnomADg_REMAINING_AF, gnomADg_SAS_AF, MAX_AF, MAX_AF_POPS, CLIN_SIG, SOMATIC, PHENO, PUBMED, MOTIF_NAME, MOTIF_POS, HIGH_INF_POS, MOTIF_SCORE_CHANGE, TRANSCRIPTION_FACTORS]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 82 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_likely_patho = df[df[\"CLIN_SIG\"].str.contains(\"likely_pathogenic\", case=False, na=False)]\n",
    "df_likely_patho"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "31fce2c1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variants with pathogenicity: 1\n"
     ]
    }
   ],
   "source": [
    "print(\"Variants with pathogenicity:\", df_patho[\"Uploaded_variation\"].nunique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "accec8d0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variants with likely pathogenicity: 0\n"
     ]
    }
   ],
   "source": [
    "print(\"Variants with likely pathogenicity:\", df_likely_patho[\"Uploaded_variation\"].nunique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "843a2e95",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1     22067\n",
       "19    18625\n",
       "11    18059\n",
       "17    15733\n",
       "2     14747\n",
       "3     13761\n",
       "12    11830\n",
       "6     11408\n",
       "4     10548\n",
       "16    10547\n",
       "5     10295\n",
       "7      9355\n",
       "10     9150\n",
       "14     8390\n",
       "9      8352\n",
       "15     7696\n",
       "8      6101\n",
       "22     5717\n",
       "20     4435\n",
       "13     3821\n",
       "18     2926\n",
       "21     2450\n",
       "X      2040\n",
       "Y         2\n",
       "Name: chr, dtype: int64"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#No. of variants per chromosome\n",
    "df[['chr', 'position']] = df['Location'].str.split(':', expand=True)\n",
    "df[\"chr\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "5e3b48ec",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variants without rsIDs:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "30726"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Identify rows that do NOT start with 'rs'\n",
    "no_rs = df[~df['Uploaded_variation'].str.startswith('rs', na=False)]\n",
    "\n",
    "# Optional: rows with rsIDs\n",
    "with_rs = df[df['Uploaded_variation'].str.startswith('rs', na=False)]\n",
    "\n",
    "# Print or save\n",
    "print(\"Variants without rsIDs:\")\n",
    "no_rs[\"Uploaded_variation\"].nunique()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "4b5c6c00",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Variants with rsIDs:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(\"Variants with rsIDs:\")\n",
    "with_rs[\"Uploaded_variation\"].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "480baee6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "qiime2-amplicon-2023.9",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
